MainModule/Server/Core/Admin.lua

```diff
@@ -1079,56 +1065,65 @@ return function(Vargs, GetEnv)
 			return (lvl and lvl.Level) or tonumber(str)
 		end;
 
-		CheckComLevel = function(plrAdminLevel, comLevel)
+		CheckComLevel = function(plrAdminLevel: number, comLevel: string | number | {[any]: number})
 			if type(comLevel) == "string" then
 				comLevel = Admin.StringToComLevel(comLevel)
 			end
 
 			if type(comLevel) == "number" and plrAdminLevel >= comLevel then
 				return true;
 			elseif type(comLevel) == "table" then
-				for i,level in pairs(comLevel) do
+				for _,level in pairs(comLevel) do
 					if plrAdminLevel == level then
 						return true
 					end
 				end
 			end
 		end;
 
-		IsBlacklisted = function(p)
+		IsBlacklisted = function(p: Player)
 			local CheckTable = Admin.CheckTable
-			for i,list in pairs(Variables.Blacklist.Lists) do
+			for _,list in pairs(Variables.Blacklist.Lists) do
 				if CheckTable(p, list) then
 					return true
 				end
 			end
 		end;
 
-		CheckPermission = function(pDat, cmd)
+		CheckPermission = function(pDat: {[string]: any}, cmd: {[string]: any})
 			local adminLevel = pDat.Level
 			local isDonor = (pDat.isDonor and (Settings.DonorCommands or cmd.AllowDonors))
+
 			local comLevel = cmd.AdminLevel
+
 			local funAllowed = Settings.FunCommands
 			local crossServerAllowed = Settings.CrossServerCommands
 
+			--// Creators rank will bypass any permissions set
 			if adminLevel >= Settings.Ranks.Creators.Level then
 				return true
+			elseif (comLevel == "Players" or comLevel == 0) and not Settings.PlayerCommands then
+				--// If PlayerCommands is set to false it will prevent users from using "Players" level commands
+				return false
 			elseif cmd.Fun and not funAllowed then
+				--// If FunCommands are disabled it will block any command Labeled "Fun"
 				return false
 			elseif cmd.IsCrossServer and not crossServerAllowed then
+				--// If CrossServerCommands is disabled it will block any command that is "CrossServer"
 				return false
 			elseif cmd.Donors and isDonor then
-				return true
-			elseif comLevel == 0 and Settings.PlayerCommands then
```

Comment:
This seems like this would break existing functionality? Has this bit been tested properly?

Reply:
Disregard. This is taken from #750. I trust Fxe knows what he's doing.